test/tests.py

```diff
@@ -831,6 +831,29 @@ def test(self):
             raise TestFail(f'stats file not generated')
 
 
+class SizeStatsTestCase(CTestCase):
+    '''Test that the DMLC_DUMP_INPUT_FILES variable works.  It creates a
+    tarball of input DML files that can be compiled standalone.'''
+    __slots__ = ()
+    def test(self):
+        super().test()
+        stats = json.loads(
+            (Path(self.scratchdir)
+             / f'T_{self.shortname}-size-stats.json').read_text())
+        top = [tuple(x) for x in stats[:4]]
+        self.pr(f'{top}')
+        # test premises: m1..m4 made it to top 4
+        assert all('size_stats.dml' in entry[2] for entry in top)
+        assert top[1][0] > 50000
+        assert {t[1] for t in top[:2]} == {1, 4}
+        assert [t[1] for t in top[2:]] == [1, 1]
+        assert top[3][0] > 10000
+        assert top[2][0] < 30000
+        by_lineno = sorted(top, key=lambda t: int(t[2].rsplit(':')[-2]))
+        # the largest ones are the non-shared, i.e. second and fourth
+        assert {by_lineno[1], by_lineno[3]} == {top[0], top[1]}
```

Comment:
This test is rather fast and loose ( _especially_ finding the non-shared methods through sorting by line numbers, _wow_ that's hacky) but I'd still say it's fine.

Reply:
What worries me a bit about the test is that it's inconcise, hard to understand even myself what I'm trying to test: What I really wanted to test is that data in the json have correct types and values in the right ballpark, then I do that by writing toy code that imitates the optimizations I envision, and then it feels like I end up validating that some very specific DML code yields C code within very specific size bounds, which isn't really what I meant.

I think I'll just change the `top = data[:4]` part and take `[x for x in data if 'size_stats' in x[2]]` instead, to allow shrinking the non-exploded variants, allowing looser bounds.